ql/pricingengines/capfloor/blackcapfloorengine.cpp

```diff
@@ -92,11 +95,13 @@ namespace QuantLib {
             // should be implemented.
             // For the time being just discard expired caplets
             if (paymentDate > settlement) {
-                DiscountFactor d = arguments_.nominals[i] *
+                Real accrualFactor = arguments_.nominals[i] *
                                    arguments_.gearings[i] *
-                                   discountCurve_->discount(paymentDate) *
                                    arguments_.accrualTimes[i];
-
+                DiscountFactor d = accrualFactor *
```

Comment:
This is not a discount factor either (only the second factor is).  "discountedAccrual", maybe?

Reply:
You are right, that that it is not a discount factor. Before my change, discountedAccrual would be a good name. As you see, I added multiplication by notional and I collect that in a vector of `deflators`. In order to avoid problems, I changed the type to `Real`.

Reply:
Now there should be no confusion.